scrapy_fake_useragent/providers.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+from fake_useragent import UserAgent
+from faker import Faker
+
+logger = logging.getLogger(__name__)
+
+class ProvidersBase:
+    '''
+    Base class for providers. Doesn't provide much functionality for now,
+    but it is a good placeholder for future additions
+    '''
+    def __init__(self, settings):
+        self.fallback = fallback
+        self.settings = settings
+
+        self._ua_type = None    # Each provider should set their own type of UA
+
+    def get_random_ua(self):
+        raise NotImplementedError
+
+
+class FixedUserAgent(ProvidersBase):
+    '''
+    Provided a fixed UA string, specified in Scrapy's settings.py
+    '''
+    def __init__(self, settings):
+        super().__init__(settings)
+
+        fixed_ua = settings.get('USER_AGENT', None)
+
+        # If the USER_AGENT setting is not set, the useragent will be empty
+        self._ua = fixed_ua if fixed_ua else ''
```

Comment:
`fixed_ua or ''` shorthand could be used

Reply:
Didn't get this one. Can you put the full snippet?

Reply:
Sure, `self._ua = fixed_ua or ''`

Reply:
Didn't know that thing. Will do it.

Anyway I wanted to point out that even though it's shorter, seems dangerous. I would tend to think that it's just an `or`, but it is a masked ternary operator. I looks to much like JS, ha.